app/src/main/java/com/example/chickenmasala/ui/screen/GuessTheGameFragment.kt

```diff
@@ -0,0 +1,178 @@
+package com.example.chickenmasala.ui.screen
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import com.bumptech.glide.Glide
+import com.example.chickenmasala.R
+import com.example.chickenmasala.data.CsvDataSource
+import com.example.chickenmasala.data.domain.GuessGamesName
+import com.example.chickenmasala.data.domain.QuestionGames
+import com.example.chickenmasala.data.interactors.GuessGamesInteractor
+import com.example.chickenmasala.data.utils.RecipeParser
+import com.example.chickenmasala.databinding.FragmentGuessTheMealBinding
+import com.example.chickenmasala.util.Constants
+import kotlinx.coroutines.*
+
+class GuessTheGameFragment(private val gameName: GuessGamesName) :
+    BaseFragment<FragmentGuessTheMealBinding>() {
+    override val LOG_TAG: String = "GuessTheMealFragment"
+    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentGuessTheMealBinding =
+        FragmentGuessTheMealBinding::inflate
+    private lateinit var guessTheMeal: QuestionGames
+    private var selectedChoiceId: Int = 0
+    private var yellowColor: Int? = null
+    private var grayColor: Int? = null
+    private var greenColor: Int? = null
+    private var redColor: Int? = null
+    private var multiChoices: List<TextView>? = null
+    override fun setup() {
+        val recipeParser = RecipeParser()
+        val dataSource =
+            CsvDataSource(requireContext(), Constants.RECIPES_CSV_FILE_NAME, recipeParser)
+        guessTheMeal = GuessGamesInteractor(dataSource).guessGames(gameName)
+        multiChoices =
+            listOf(
+                binding.answerOneText,
+                binding.answerTwoText,
+                binding.answerThreeText,
+                binding.answerFourText
+            )
+        showGame()
+    }
+
+    private fun showGame() {
+        prepareColors()
+        prepareMultiChoices()
+        prepareQuestion()
+        setButtonStatus(binding.submitButton, false)
+        prepareNextQuestion(clickChoicesState = true)
+    }
+
+    override fun addCallBacks() {
+        setColorOfSelectedChoice()
+        submitAnswer()
+    }
+
+    private fun prepareColors() {
+        yellowColor = ContextCompat.getColor(requireContext(), R.color.yellow_600)
+        grayColor = ContextCompat.getColor(requireContext(), R.color.gray_200)
+        greenColor = ContextCompat.getColor(requireContext(), R.color.green_100)
+        redColor = ContextCompat.getColor(requireContext(), R.color.red_100)
+        setUpChoicesColor()
+    }
+
+    private fun setUpChoicesColor() {
+        multiChoices?.forEach { choice ->
+            colorizeTheView(choice, grayColor!!)
```

Comment:
use let scope function instead of not-null assertion operator because this operation danger